### PR TITLE
Adds GitHub App based auth

### PR DESCRIPTION
**Pull request by tcsc**
Adds Github App-based authentication to the workflow runner script.

### CODE REVIEW

**Review by greedy52 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

File: build.assets/tooling/cmd/gh-trigger-workflow/args.go [hunk 1]
```diff
@@ -51,9 +52,22 @@ func (m inputMap) Set(s string) error {
 	return nil
 }
 
+// key holds the bytes of a GutHub app key, stored as an array of bytes.
```
Comment (zmb3) [inline build.assets/tooling/cmd/gh-trigger-workflow/args.go:55]:
```suggestion
// key holds the bytes of a GitHub app key
```

Technically it's a slice, not an array. But that parts obvious anyway, so let's just omit it.

**Review by zmb3 — APPROVED** [review]

**Review by camscale — APPROVED** [review]

File: build.assets/tooling/cmd/gh-trigger-workflow/main.go [hunk 2]
```diff
@@ -39,35 +39,50 @@
...
 package main
 
 import (
 	"context"
+	"flag"
 	"log"
+	"net/http"
 	"time"
 
+	ghinst "github.com/bradleyfalzon/ghinstallation"
```
[the remark this answers is not in the document]
Comment (tcsc) [reply build.assets/tooling/cmd/gh-trigger-workflow/main.go]: Updated

File: build.assets/tooling/cmd/gh-trigger-workflow/args.go (hunk 1 above)
Comment (tcsc) [reply build.assets/tooling/cmd/gh-trigger-workflow/args.go]: Added comments as to why we care that its in a slice rather than a string.

Comment (zmb3) [reply build.assets/tooling/cmd/gh-trigger-workflow/args.go]: But it still says `GutHub` not `GitHub`.

Comment (camscale) [reply build.assets/tooling/cmd/gh-trigger-workflow/args.go]: it's ok. guthub.com works too :trollface:

**Pull request merged**